bugsnag-android-core/src/main/java/com/bugsnag/android/ThreadState.kt

```diff
@@ -29,49 +28,55 @@ internal class ThreadState @Suppress("LongParameterList") @JvmOverloads construc
             (sendThreads == ThreadSendPolicy.UNHANDLED_ONLY && isUnhandled)
 
         threads = when {
-            recordThreads -> captureThreadTrace(
-                stackTraces ?: java.lang.Thread.getAllStackTraces(),
-                currentThread ?: java.lang.Thread.currentThread(),
-                exc,
-                isUnhandled,
-                projectPackages,
-                logger
-            )
+            recordThreads -> captureThreadTrace(exc, isUnhandled, maxThreads, projectPackages, logger)
             else -> mutableListOf()
         }
     }
 
     private fun captureThreadTrace(
-        stackTraces: MutableMap<java.lang.Thread, Array<StackTraceElement>>,
-        currentThread: java.lang.Thread,
         exc: Throwable?,
         isUnhandled: Boolean,
+        maxThreadCount: Int,
         projectPackages: Collection<String>,
         logger: Logger
     ): MutableList<Thread> {
-        // API 24/25 don't record the currentThread, add it in manually
-        // https://issuetracker.google.com/issues/64122757
-        if (!stackTraces.containsKey(currentThread)) {
-            stackTraces[currentThread] = currentThread.stackTrace
-        }
-        if (exc != null && isUnhandled) { // unhandled errors use the exception trace for thread traces
-            stackTraces[currentThread] = exc.stackTrace
-        }
+        val currentThread = java.lang.Thread.currentThread()
+        val threadCount = java.lang.Thread.activeCount()
+        val wantedThreadCount = kotlin.math.min(threadCount, maxThreadCount)
+        val threads: Array<java.lang.Thread?> = arrayOfNulls(wantedThreadCount)
+        val recordedThreadCount = java.lang.Thread.enumerate(threads)
```

Comment:
`Thread.enumerate` will only report threads in our current thread-group, while `getAllStackTraces` will list all threads in this apps runtime.

To get all of the process threads, we still need to find the system `ThreadGroup`:
```kotlin
private fun rootThreadGroup(): ThreadGroup {
    var group = Thread.currentThread().threadGroup
    
    while(group.parent != null) {
        group = group.parent
    }
    
    return group
}
```